vignettes/cgdsMigration.Rmd

```diff
@@ -17,149 +18,295 @@ output:
-CRAN package. Note that the `cgds` package code is shown for comparison but it
+This vignette aims to help developers migrate from the now defunct `cgdsr`
+CRAN package. Note that the `cgdsr` package code is shown for comparison but it
 is not guaranteed to work. If you have questions regarding the contents,
-please create an issue at the GitHub repository:
-https://github.com/waldronlab/cBioPortalData/issues
+please create an issue at the [GitHub](https://github.com/waldronlab/cBioPortalData/issues) repository.
+
+
+# Loading the packages
+
+```{r, warning=FALSE, comment=FALSE, message=FALSE}
+## Import needed libraries
+if (!require("devtools")) install.packages("devtools")
+if (!require("BiocManager")) install.packages("BiocManager")
+if (!require("cgdsr")) devtools::install_version("cgdsr",version="1.3")
+#if (!require("cgdsr")) devtools::install_github("kmezhoud/cgdsr")
+if (!require("cBioPortalData")) BiocManager::install("cBioPortalData")
+#if (!require("cBioPortalData")) devtools::install_version("cBioPortalData",version="2.7.11")
+if (!require("dplyr")) install.packages("dplyr")
+if (!require("tidyr")) install.packages("tidyr")
+if (!require("DT")) install.packages("DT")
+if (!require("tictoc")) install.packages("tictoc")
 
-# Loading the package
 
-```{r,include=TRUE,results="hide",message=FALSE,warning=FALSE}
-library(cBioPortalData)
 ```
 
-# Discovering studies
+# Discovering studies {.tabset .tabset-fade .tabset-pills}
 
-## cgds
+## `cgdsr`
+```{r, comment=FALSE }
+cgds <- cgdsr::CGDS("http://www.cbioportal.org/")
+
+getCancerStudies.CGDS(cgds) %>%
+    datatable(options = list(pageLength = 5))
 
-```{r,eval=FALSE}
-library(cgdsr)
-cgds <- CGDS("http://www.cbioportal.org/")
-getCancerStudies.CGDS(cgds)
 ```
 
-## cBioPortalData
+## `cBioPortalData`
 
-Here we show the default inputs to the cBioPortal function for clarity.
+* Here we show the default inputs to the `cBioPortal` function for clarity.
 
-```{r}
+```{r , comment=FALSE, warning=FALSE, message=FALSE}
 cbio <- cBioPortal(
     hostname = "www.cbioportal.org",
     protocol = "https",
     api. = "/api/api-docs"
 )
-getStudies(cbio)
+
+getStudies(cbio) %>% 
+    select(studyId, everything()) %>%
+    datatable(options = list(pageLength = 5))
+
 ```
 
-Note that the `studyId` column is important for further queries.
+
+**Note**
+
+* The `studyId` column is important for further queries.
+
+# Obtaining Cases/Patients Identifiers {.tabset .tabset-fade .tabset-pills}
+
+**Definition**
+
+* The Cases is synonym to the Patients. 
+
+* Each patient is identified  by `Case_id`.
+
+* Each patient has one or multiple samples used for molecular/genetic  profile analysis, like mRNA, mutation, CNA, and Methylation.
+
+## `cgdsr`
 
 ```{r}
-head(getStudies(cbio)[["studyId"]])
+
+getCaseLists.CGDS(x = cgds,
+                  cancerStudy = "gbm_tcga_pub") %>% 
+    datatable(options = list(pageLength = 5))
+
 ```
+**Note**
 
-# Obtaining Cases and Clinical data
```

Comment:
Do not delete headers and keep the structure of the original vignette for easy lookup.